### PR TITLE
VE-219 Migrate RUI to MUI v6

### PR DESCRIPTION
**Pull request by m4manjesh**
Closes [VE-219](https://sce.myjetbrains.com/youtrack/issue/VE-219/Migrate-RUI-from-MUI-v5-to-v6)

This PR focuses on migrating RUI from MUI v5 to v6. There are few breaking changes based on this [documentation](https://mui.com/material-ui/migration/upgrade-to-v6/). There will be separate PR's to handle deprecations, migrating to Pigment CSS(if we prefer) etc. Some major changes introduced and updated in this PR are as follows

1. Introduction of `cssVariables`  flag, to enable access to css variables from the `theme.vars` object, [read more](https://mui.com/blog/material-ui-v6-is-out/#css-theme-variables). 
2. Introduction of `colorSchemes` node to simplify the process to support light and dark mode, [read more
](https://mui.com/blog/material-ui-v6-is-out/#color-schemes)
3. Introduction of new API `theme.applyStyles()` to replace the `theme.palette.mode === 'dark'` condition. This now solves the [dark mode flickering issue](https://github.com/mui/material-ui/issues/27651) when combined with the CSS theme variables feature above.
4.  `Grid` is deprecated , importing it as `Grid2` from `'@mui/material/Grid2'`  or `Grid2 as Grid`  from `"@mui/material"`, [read more](https://mui.com/material-ui/migration/upgrade-to-v6/#grid2)
5. The `color` prop in the `Typography` component is not a [system prop](https://mui.com/system/properties/) anymore. Updated to use as `sx` prop now, [read more](https://mui.com/material-ui/migration/upgrade-to-v6/#typography)

## Changes

**Package Updates**
- Updated `"@mui/material"` and `"@mui/icons-material"` to `6.0.0`

**Accessing theme**
- Accessing theme inside `sx` prop changes in this PR as follows. There is no harm following the old usage but we are updating this for future protection. 
Screenshot 2024-11-26 105327

**Accessing mode**
- A new method was added throughout the app to apply styles for dark and light modes
Screenshot 2024-11-26 105657

**Grid Updates**
- As mentioned above `Grid` is deprecated and we use `[Grid2](https://mui.com/material-ui/migration/upgrade-to-v6/#grid2)` now and props like `xs, sm, md, lg, xl` and `xsOffset, smOffset, mdOffset, lgOffset, xlOffset` renamed as `Size` and `Offset`. Usage as follows
Screenshot 2024-11-26 110151

**Removing a few deprecations**
- A few `TextField` were updated to use `slotProps` instead of the deprecated `inputProps`. As this PR grows, I plan to handle all other deprecations separately. 
Screenshot 2024-11-26 110808

**Spell checks**
- Spelling mistakes noticed and corrected in `PasswordChangeDialog.jsx`

**Typography**
- Removed using `color` as system prop, instead using as follows
Screenshot 2024-11-26 111911

## UI/UX

N/A

## Testing notes

There are no new visible UI changes, but as this is a major package update, I suggest checking that all components still work fine as before.

## Author checklist

Before I request a review:

- [x] I have reviewed my own code-diff.
- [x] I have tested the changes in Docker / a deploy-preview.
- [x] I have assigned the PR to myself or an appropriate delegate.
- [x] I have added the relevant labels to the PR.
- [x] I have included appropriate tests.
- [x] I have checked that the Lint and Test workflows pass on Github.
- [ ] ~I have populated the deploy-preview with relevant test data.~
- [ ] ~I have tagged a UI/UX designer for review if this PR includes UI/UX changes.~

### CODE REVIEW

**Comment by nikitharani** [comment]
1) We have some components where tests use fireEvent.click(button). No React warnings are being logged during tests. Since it is mentioned as a breaking change in the documentation, updating the tests ensures compliance with MUI v6 updated behavior. 
https://mui.com/material-ui/migration/upgrade-to-v6/#breaking-changes-affecting-testing
APP Layout mode

2) VirtoAppLayout
Before
VirtoAppLayout-Main

After
VirtoAppLayout-after

**Comment by m4manjesh** [comment]
> 1. We have some components where tests use fireEvent.click(button). No React warnings are being logged during tests. Since it is mentioned as a breaking change in the documentation, In my opinion updating the tests ensures compliance with MUI v6 updated behavior.

No tests are failing now so we dont need to update this, confirmed with @mattcorner

**Comment by m4manjesh** [comment]
> 2\. VirtoAppLayout

I need to combine the current seperate themse variables we have for dark and white into one as MUI suggests [here ](https://mui.com/material-ui/customization/palette/#color-schemes). Also planning to handle this in a seperate PR.

File: src/ThemeProvider/ThemeProvider.tsx
Comment (mattcorner) [inline src/ThemeProvider/ThemeProvider.tsx:1]: I thought we agreed to back out of these changes for now and deal with them seperately.

File: package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
   "devDependencies": {
     "@emotion/react": "^11.11.4",
     "@emotion/styled": "^11.11.5",
-    "@mui/material": "^5.16.4",
+    "@mui/material": "^6.0.0",
```
Comment (mattcorner) [inline package.json:37]: Why just 6.0.0? Why not 6.1.9 which is latest?

File: src/DeletableList/DeletableList.tsx [hunk 2]
```diff
@@ -13,7 +13,9 @@ const DeletableList = ({
       {items?.map(value => {
         return (
           <ListItem
-            sx={{ py: theme => theme.spacing(1) }}
+            sx={theme => ({
+              py: theme.spacing(1)
+            })}
```
Comment (mattcorner) [inline src/DeletableList/DeletableList.tsx:18]:
```suggestion
            sx={{ py: 1 }}
```

**Review by mattcorner — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (m4manjesh) [reply package.json:37]: Yes we can update to 6.1.9, i was following what documentation suggests [here](https://mui.com/material-ui/migration/upgrade-to-v6/#quality-of-life-improvements)

**Review by mattcorner — COMMENTED** [review]
I've done no testing, but the code changes look fine to me. @nikitharani to complete the full review.

**Review by nikitharani — APPROVED** [review]
LGTM

**Pull request merged**